### PR TITLE
[CS-3863]: Adds PinScreen UI

### PR DESCRIPTION
**Pull request by danibonilha**
### Description
 This PR creates the PINScreen UI, the goal is to be able to reuse it across the most flows, there's some hardcorded strings and params for now, just to show functionality, but the implementation to be fully generic is going to be done on the logic ticket it also adds shareable dark | light styles, updates biometric, input and cell to use it and the ability to show/hide the pin 

### Screenshots

image

image

image

image

// up-to-date info icon
image

### CODE REVIEW

File: cardstack/src/navigation/screens.ts [hunk 1]
```diff
@@ -207,6 +208,10 @@ export const MainScreens: Record<keyof typeof MainRoutes, ScreenNavigation> = {
     component: ColorPickerModal,
     options: expandedPreset as StackNavigationOptions,
   },
+  PIN_SCREEN: {
+    component: PinScreen,
+    options: horizontalInterpolator,
```
Comment (oidoug) [inline cardstack/src/navigation/screens.ts:213]: `horizontalInterpolator` allows for horizontal gestures to close the screen on iOS.

Maybe we can discuss a new appear animation for the PIN screen too.

**Review by paulinhapenedo — APPROVED** [review]
LGTM 🚀

File: cardstack/src/screens/PinScreen/PinScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                color={colorStyleVariants.textColor[variant]}
+                textAlign="center"
+              >
+                {feedbackProps.label}
+              </Text>
+              <Icon
+                name={feedbackProps.iconName}
+                size={20}
+                paddingLeft={2}
+                backgroundColor="white"
```
Comment (oidoug) [inline cardstack/src/screens/PinScreen/PinScreen.tsx:106]: shouldn't this follow colorStyleVariants too?

**Review by oidoug — APPROVED** [review]
LGTM, I just added some comments for later evaluation.

File: cardstack/src/screens/PinScreen/PinScreen.tsx (hunk 2 above)
Comment (danibonilha) [reply cardstack/src/screens/PinScreen/PinScreen.tsx:106]: This is the color of the ! point, and on both themes the color is the same

File: cardstack/src/navigation/screens.ts (hunk 1 above)
Comment (danibonilha) [reply cardstack/src/navigation/screens.ts:213]: just a placeholder for now, once we build the flows we can decide the correct animations

**Pull request merged**